devimint/src/tests.rs

```diff
@@ -350,6 +350,151 @@ pub async fn latency_tests(dev_fed: DevFed, r#type: LatencyTest) -> Result<()> {
+                    fm_port_gw_lnd = &process_mgr.globals.FM_PORT_GW_LND,
+                    gateway_cli = crate::util::get_gateway_cli_path().join(" ")
+                );
+                std::env::set_var("FM_GWCLI_CLN", fm_gwcli_cln);
+                std::env::set_var("FM_GWCLI_LND", fm_gwcli_lnd);
+            };
+
+            anyhow::ensure!(!paths.is_empty(), "Must provide at least 1 binary path");
+            let oldest_gateway_cli = paths.first().expect("at least 1 binary was provided");
+            update_gateway_cli_path(oldest_gateway_cli);
+            let gateway_cli_version = crate::util::GatewayCli::version_or_default().await;
+
+            info!(
+                "running first stress test for gateway-cli version: {}",
+                gateway_cli_version
+            );
+            let dev_fed = dev_fed(process_mgr).await?;
+            let client = dev_fed.fed.new_joined_client("test-client").await?;
+            try_join!(stress_test_fed(&dev_fed), wait_session(&client))?;
+
+            // add the current gateway-cli binary from PATH
+            paths.push("gateway-cli".into());
+            for path in paths.iter().skip(1) {
+                update_gateway_cli_path(path);
+                let gateway_cli_version = crate::util::GatewayCli::version_or_default().await;
+                info!("upgraded gateway-cli to version: {}", gateway_cli_version);
+                try_join!(stress_test_fed(&dev_fed), wait_session(&client))?;
+                info!(
+                    "### gateway-cli passed stress test for version {}",
+                    gateway_cli_version
+                );
+            }
+            info!("## gateway-cli upgraded all binaries successfully")
+        }
+        UpgradeTest::Gatewayd { mut paths } => {
+            anyhow::ensure!(!paths.is_empty(), "Must provide at least 1 binary path");
+            let oldest_gatewayd = paths.first().expect("at least 1 binary was provided");
+            std::env::set_var("FM_GATEWAYD_BASE_EXECUTABLE", oldest_gatewayd);
+            let gatewayd_version = crate::util::Gatewayd::version_or_default().await;
+
+            info!(
+                "running first stress test for gatewayd version: {}",
+                gatewayd_version
+            );
+            let mut dev_fed = dev_fed(process_mgr).await?;
+            let client = dev_fed.fed.new_joined_client("test-client").await?;
+            try_join!(stress_test_fed(&dev_fed), wait_session(&client))?;
+
+            // add the current gatewayd binary from PATH
+            paths.push("gatewayd".into());
+            for path in paths.iter().skip(1) {
+                try_join!(
+                    dev_fed.gw_cln.restart_with_bin(process_mgr, path),
+                    dev_fed.gw_lnd.restart_with_bin(process_mgr, path),
+                )?;
+                try_join!(stress_test_fed(&dev_fed), wait_session(&client))?;
```

Comment:
what restarted the gateways while the stress tests were running, 

something like
```rust
try_join!(stress_test_fed(&dev_fed), async { sleep(30s).await; upgrade_gws(); })
```

Reply:
Capturing our chat on the dev call: this is a bit tricky to restart while running latency tests and can be addressed later.